src/python/pants/backend/python/util_rules/pex_venv.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import Path
+
+from pants.backend.python.util_rules import pex_cli
+from pants.backend.python.util_rules.pex import CompletePlatforms, Pex, PexPlatforms
+from pants.backend.python.util_rules.pex_cli import PexCliProcess
+from pants.engine.fs import Digest
+from pants.engine.internals.native_engine import MergeDigests
+from pants.engine.process import ProcessResult
+from pants.engine.rules import Get, collect_rules, rule
+
+
+class PexVenvLayout(Enum):
+    VENV = "venv"
+    FLAT = "flat"
+    FLAT_ZIPPED = "flat-zipped"
+
+
+@dataclass(frozen=True)
+class PexVenvRequest:
+    pex: Pex
+    layout: PexVenvLayout
+    output_path: Path
+    description: str
+
+    platforms: PexPlatforms = PexPlatforms()
+    complete_platforms: CompletePlatforms = CompletePlatforms()
+
+
+@dataclass(frozen=True)
+class PexVenv:
```

Comment:
I suspect this name will be confusing with `VenvPex`, but I lost inspiration, let me know if you have a better name.